src/tools/tests/storage-key-recipe-resolver-test.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Manifest} from '../../runtime/manifest.js';
+import {assert} from '../../platform/chai-node.js';
+import {StorageKeyRecipeResolver} from '../storage-key-recipe-resolver.js';
+
+describe('recipe2plan', () => {
+  it('Long + Long: If ReadingRecipe is long running, it is a valid use case', async () => {
+    const manifest = await Manifest.parse(`\
+    particle Reader
+      data: reads Thing {name: Text}
+
+    // TODO(alxr): Resolve these types later
+    // particle Writer
+    //   data: writes Product Thing {name: Text, price: Number}
+
+    particle Writer
+       data: writes Thing {name: Text}
+    
+    @trigger
+      launch startup
+      arcId myArcId
+    recipe WritingRecipe
+      thing: create persistent 'my-handle-id' 
+      Writer
+        data: writes thing
+
+    @trigger
+      launch startup
+      arcId otherArcId
+    recipe ReadingRecipe
+      data: map 'my-handle-id'
+      Reader
+        data: reads data`);
+
+    const resolver = new StorageKeyRecipeResolver(manifest);
+    for await (const it of resolver.resolve()) {
+      assert.isTrue(it.isResolved());
```

Comment:
at some later stage it would be nice to ensure storage keys of the mapped and created handle are equal, but that may be easier once you flesh out more.

Reply:
Yeah, definitely!